client/src/components/App.js

```diff
@@ -1,70 +1,72 @@
-import React, { Component } from 'react';
-import NavBar from './NavBar';
-import Temperature from './Temperature'
-import Login from './Login';
-import 'bootstrap/dist/css/bootstrap.min.css';
-import '../css/App.css';
-import { appConfig } from '../utils/constants'
-import { UserSession } from 'blockstack';
-import Button from 'react-bootstrap/Button';
-import { configure, User, getConfig } from 'radiks';
-import Patient from '../models/patient'
+import React, { Component } from "react";
+import NavBar from "./NavBar";
+import Temperature from "./Temperature";
+import Login from "./Login";
+import "bootstrap/dist/css/bootstrap.min.css";
+import "../css/App.css";
+import { appConfig } from "../utils/constants";
+import { UserSession } from "blockstack";
+import Button from "react-bootstrap/Button";
+import { configure, User, getConfig } from "radiks";
+import Patient from "../models/patient";
 
 const userSession = new UserSession({
   appConfig: appConfig
-})
+});
 
 export default class App extends Component {
-
   constructor(props) {
-    super(props)
-    this.userSession = new UserSession({ appConfig })
+    super(props);
+    this.userSession = new UserSession({ appConfig });
   }
 
   async componentDidMount() {
+    if (!process.env.REACT_APP_QA_URL) {
+      return;
+    }
     configure({
       apiServer: process.env.REACT_APP_QA_URL, // TODO this will change to wherever our radiks server will be hosted in prod
-      userSession: this.userSession,
+      userSession: this.userSession
     });
-    const { userSession } = getConfig();
-    if (userSession.isSignInPending()) {
-      await userSession.handlePendingSignIn();
-      await User.createWithCurrentUser();
-      window.location = '/';
-    } else if (userSession.isUserSignedIn()) {
 
+    const { userSession } = getConfig();
+    if (userSession.isUserSignedIn()) {
       // Creates a new Patient model associated with the user
       const patient = new Patient({
         doctor: "Test Doctor",
-        location: ['123', '456'],
-      })
+        location: ["123", "456"]
+      });
 
       // Saves that patient in the user's associated Gaia storage, encrypted, and replicated in MongoDB
-      await patient.save()
+      await patient.save();
 
       // Radiks queries the encrypted MongoDB entry, decrypts the data
-      const allPatients = await Patient.fetchOwnList()
+      const allPatients = await Patient.fetchOwnList();
 
-      // Print the resulting 
-      console.log('ALL PATIENTS:', allPatients)
+      // Print the resulting
+      console.log("ALL PATIENTS:", allPatients);
 
       // Delete the entry, to keep things clean for the purpose of example
       var p;
       for (p of allPatients) {
-        p.destroy()
+        p.destroy();
       }
+    } else if (userSession.isSignInPending()) {
+      await userSession.handlePendingSignIn();
+      await User.createWithCurrentUser();
+      window.location = "/";
```

Comment:
I'm just curious because maybe you've worked with Blockstack longer than I have (see: < 1 week for me) - is there any strict reasoning to put `isSignInPending()` after checking `isUserSignedIn()`? Outside of maybe slightly faster performance (which is always a plus), is it more just a code clarity thing?

Reply:
it is about the way that `isSignInPending` is resolved: the function just looks at the url query for `authResponse`.
Therefore, it is possible the the user is signed in and sign in is pending if the url query parameter `authResponse` is not removed.

Reply:
Ah, that makes a lot of sense - thanks!